webviz_subsurface/private_containers/_surface_selector.py

```diff
@@ -0,0 +1,586 @@
+        self.attr_id_btn_prev = f"{uuid}-attr-btn-prev"
+        self.attr_id_btn_next = f"{uuid}-attr-btn-next"
+        self.name_id = f"{uuid}-name"
+        self.name_id_btn_prev = f"{uuid}-name-btn-prev"
+        self.name_id_btn_next = f"{uuid}-name-btn-next"
+        self.date_id = f"{uuid}-date"
+        self.date_id_btn_prev = f"{uuid}-date-btn-prev"
+        self.date_id_btn_next = f"{uuid}-date-btn-next"
+        self.ensemble_id = f"{uuid}-ens"
+        self.ensemble_id_btn_prev = f"{uuid}-ensemble-btn-prev"
+        self.ensemble_id_btn_next = f"{uuid}-ensemble-btn-next"
+        self.realization_id = f"{uuid}-real"
+        self.realization_id_btn_prev = f"{uuid}-realization-btn-prev"
+        self.realization_id_btn_next = f"{uuid}-realization-btn-next"
+        self.attr_wrapper_id = f"{uuid}-attr-wrapper"
+        self.name_wrapper_id = f"{uuid}-name-wrapper"
+        self.date_wrapper_id = f"{uuid}-date-wrapper"
+        self.ens_wrapper_id = f"{uuid}-ens-wrapper"
+        self.real_wrapper_id = f"{uuid}-real-wrapper"
+        self.aggreal_id = f"{uuid}-aggreal"
+        self.sens_name_id = f"{uuid}-sens-name-id"
+        self.sens_case_id = f"{uuid}-sens-case-id"
+        self.sens_name_wrapper_id = f"{uuid}-sens-name-wrapper-id"
+        self.sens_case_wrapper_id = f"{uuid}-sens-case-wrapper-id"
+
+    @property
+    def attrs(self):
+        return list(self._configuration.keys())
+
+    def names_in_attr(self, attr):
+        return self._configuration[attr].get("names", None)
+
+    def dates_in_attr(self, attr):
+        return self._configuration[attr].get("dates", None)
+
+    @property
+    def ensembles(self):
+        return list(self._ensembles["ENSEMBLE"].unique())
+
+    def sens_names(self, ensemble):
+        sensnames = list(
+            self._ensembles.loc[self._ensembles["ENSEMBLE"] == ensemble][
+                "SENSNAME"
+            ].unique()
+        )
+        if sensnames[0] == None:
+            return None
+        return sensnames
+
+    def sens_cases(self, ensemble, sensname):
+        senscases = list(
+            self._ensembles.loc[
+                (self._ensembles["ENSEMBLE"] == ensemble)
+                & (self._ensembles["SENSNAME"] == sensname)
+            ]["SENSCASE"].unique()
+        )
+        if senscases and senscases[0] == None:
+            return None
+        return senscases
+
+    def realizations(self, ensemble, sensname=None, senstype=None):
+        df = self._ensembles.loc[self._ensembles["ENSEMBLE"] == ensemble].copy()
+        if sensname and senstype:
+            df = df.loc[(df["SENSNAME"] == sensname) & (df["SENSCASE"] == senstype)]
+        return list(df["REAL"])
+
+    @property
+    def aggregations(self):
+        return ["mean", "stddev", "min", "max", "p10", "p90"]
+
+    @property
+    def show_dropdown_style(self):
+        return {"display": "grid"}
+
+    @property
+    def hide_dropdown_style(self):
+        return {"visibility": "hidden"}
```

Comment:
The `@property` decorated functions with constant return value, should they instead simply be either class or instance constants instead?